ITE/4/README.adoc

```diff
@@ -0,0 +1,462 @@
+the artifact in a consistent manner. We discuss this and other sources of
+non-determinism in the Appendix.
+
+An example of the product field for link metadata with a generic URI is as
+follows:
+
+```
+"products": {
+    "github+https://api.github.com/repos/in-toto/in-toto/pulls/250": {
+        "sha256": "7ee0617..."
+    },
+    "...": ...
+}
+```
+
+Similarly, a generic URI can be used to refer to other abstract entities like an
+SPDX element. The URI can be used to directly fetch the corresponding documents
+from a specified location. For in-toto metadata that are embedded within SPDX
+documents, the URI can also be used to refer to other entities elsewhere in the
+document. Resolving such URIs that point to local entities elsewhere in the SPDX
+document is an implementation level detail - but it's clear that the URI must
+contain all the information necessary to identify the entity in that context.
+
+It is important to note that this ITE doesn't mandate any specific generic URI
+tokens that all conforming implementations MUST support. Instead, it merely
+specifies what any generic URI must look like, and what rules the system MUST
+follow.
+
+in-toto artifact rules are a means to specify what artifacts the project owner
+expects in every step of the supply chain in the in-toto layout. in-toto
+provides the following rules that together attempt to describe the different
+types of operations or actions that supply chain functionaries carry out.
+
+*MATCH Rule*
+
+`MATCH <pattern> [IN <source-path-prefix>] WITH (MATERIALS|PRODUCTS) [IN
+<destination-path-prefix>] FROM <step>`
+
+The `MATCH` rule is a convenient way to match artifacts (either in materials or
+products depending on where the rule is specified) with artifacts from other
+steps in the supply chain, allowing owners to establish a flow of artifacts
+between steps in the software supply chain.
+
+The current verification workflow for the `MATCH` rule compares the
+cryptographic hashes of the relevant artifacts from the respective link files.
+This is a straightforward comparison and does not entail resolving the URIs in
+any form. The `IN` clauses which are used to specify path prefixes are
+subtracted from the URI strings. Therefore, we conclude that the functioning of
+the rule doesn't change with this ITE. An example `MATCH` rule with generic URIs
+is as follows:
+
+`MATCH commit/* IN github+https://github.com/in-toto/in-toto/ WITH PRODUCTS IN
+github+https://github.com/in-toto/in-toto FROM merge-pull-request-250`
+
+*Other artifact rules*
+
+in-toto provides several other artifact rules - specifically `ALLOW`,
+`DISALLOW`, `CREATE`, `DELETE`, `REQUIRE`, and `MODIFY`. These rules perform
+different checks by verifying if artifacts matching the pattern occur or do not
+occur in the materials or products sections of the relevant link. This does not
+involve resolving the pattern or URI into the artifact itself, and so this ITE
+does not affect the working of these rules.
```

Comment:
I may be a little confused here, but why wouldn't it change these?  If you are creating an item at a URI, why can't in-toto restrict / validate that?

Reply:
Currently, these rules dictate that the pattern specified must appear (or not, depending on the rule) in the step or inspection's link. This is in turn dependent on the in-toto tool generating the metadata being able to record the artifacts correctly, but the verification of these rules ensures the artifact appears in the respective products / materials dictionary. This is not different depending on the type of URI - the pattern must simply appear as a key in the dictionary.

I may be misunderstanding you here, but in-toto's specification for the artifact rules are not dependent on the artifact identifiers being filepaths only, so their restrictions / validation properties should continue to hold.